### DISCUSSION TITLE
Neural Network idea?

### DISCUSSION THREAD

**Discussion by Gammer0909** [familiar/Bedwars-Bot#7] [Ideas]
There's probably a video out there explaining that someone taught an AI to play minecraft, but what if we teach an AI to play *bedwars*?

Just a thought (I'm dumb and can't AI but hmu)

**Comment by IDoTheHax** [comment]
Alr in development

**Reply by Gammer0909** [reply]
What's the repo?

**Reply by IDoTheHax** [reply]
Private, MAY be released, once public

**Reply by Gammer0909** [reply]
@IDoTheHax, but this is Open Source, we can make it stronger together!
Also, based off your idea from the other thread (People training it) it would need to be open source to be trained, and people could help you develop it, I know I would (try)

**Reply by IDoTheHax** [reply]
Whats your discord?? @Gammer0909

**Reply by Gammer0909** [reply]
@IDoTheHax I believe my Discord is @The_Game12, but: **be warned: I have not touched my disc in like 5< months**

**Reply by pixo2000** [reply]
If you need more people helping: my dc is xandarian

**Reply by IDoTheHax** [reply]
...

**Reply by RubyDiamondstar** [reply]
I like this idea lol

**Reply by Pleasntly** [reply]
> Private, MAY be released, once public

Is this released yet? I would be happy to train it if needed. I am decent at bedwars.

**Reply by IDoTheHax** [reply]
Nope, I lost the pc that had all of the code and didn't push the latest _working_ code to GitHub because I'm an idiot, 

When I get funded/save for a new computer I'll probably be able to do it

**Reply by IDoTheHax** [reply]
My discord is still IDoTheHax though

**Reply by Pleasntly** [reply]
> Nope, I lost the pc that had all of the code and didn't push the latest _working_ code to GitHub because I'm an idiot,
> 
> When I get funded/save for a new computer I'll probably be able to do it

Link to the latest push before it was working? would love to try it out if you could show me

**Reply by IDoTheHax** [reply]
never got it so restarting from scratch, i think thats better though

**Reply by Tewxx** [reply]
> Nope, I lost the pc that had all of the code and didn't push the latest _working_ code to GitHub because I'm an idiot,

looking foward to it

**Comment by familiar** [comment]
Imagine a bedwars bot that could incorporate machine learning training to build a neural network, and all of this training data is stored on a external server. As more and more people use the bot, they contribute to the training of the bot, and eventually, the bot becomes the best to ever have been made.

Just a thought.

**Reply by Gammer0909** [reply]
I like that!

It's also a little bit eerie, that the best players are feeding their demise....

**Reply by IDoTheHax** [reply]
So basically there would be two ways of training an ai to play beadwars 1. Player Pov learning: the ai would use the data collected from other players data of them playing (this would require some sort of program running on that players computer)

Then the other is Machine Learning: where we would give points based on certain things ie: kill a player, destroy a bed, collect an item, something like so

**Reply by IDoTheHax** [reply]
Which I'm alr working on, but this (data) is very valuable stuff, just this morning I lost a week's worth of model data...

**Reply by Gammer0909** [reply]
Honestly, the second way would be more efficient, as you can just set a Pi Cluster off and have it running for like a few hours and see how good it is, rather than having people run a program while they fail to play bedwars (I doubt us programmers are very good at it...)

Or, you can scrape some footage from pro players (Think Technoblade's Winstreak, etc.) and have it analyze the footage, rather than analyzing live play (that would be too much of a performance hit, I would think)

**Reply by pixo2000** [reply]
If you just need videos of wins, go on youtube. Most people only post videos if they have won the game

**Reply by Gammer0909** [reply]
> So basically there would be two ways of training an ai to play beadwars 1. Player Pov learning: the ai would use the data collected from other players data of them playing (this would require some sort of program running on that players computer)
> 
> 
> 
> Then the other is Machine Learning: where we would give points based on certain things ie: kill a player, destroy a bed, collect an item, something like so

Actually, both of those methods are Machine Learning, just different methods of getting there...

**Reply by IDoTheHax** [reply]
Yes that is true I forgot to mention that

Also accept my DC pls

**Reply by Gammer0909** [reply]
@IDoTheHax sorry forgot about this thread for a minute, what about [(Deep) Reinforcement Learning](https://en.wikipedia.org/wiki/Reinforcement_learning)? It would be an easier way to just kinda... let it run on like a pi cluster or spare PC/Laptop for like a week or two that we could then run :)

**Reply by IDoTheHax** [reply]
Sorry for the even later response lmao im currently on track to doing that, not a lot of progress since its just a test but its going somewhere

**Comment by Pleasntly** [comment]
Is this still being developed or has it been completed somewhere?
